tests/test_scalars.py

```diff
@@ -78,27 +95,37 @@ def test_period() -> None:
     as4 = int(1)
     as5 = pd.period_range("2012-1-1", periods=10, freq="D")
     as6 = pd.Period("2012-1-1", freq="D")
+    as7 = cast("TimedeltaSeries", pd.Series([pd.Timedelta(days=1)]))
+    as8 = cast("PeriodSeries", pd.Series([as6]))
 
     check(assert_type(p + as0, pd.Period), pd.Period)
     check(assert_type(p + as1, pd.Period), pd.Period)
     check(assert_type(p + as2, pd.Period), pd.Period)
     check(assert_type(p + as3, pd.Period), pd.Period)
     check(assert_type(p + as4, pd.Period), pd.Period)
-
+    check(assert_type(p + p.freq, pd.Period), pd.Period)
+    check(assert_type(p + (p - as5), pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(p + as7, "PeriodSeries"), pd.Series)
+    das8 = cast("TimedeltaSeries", (as8 - as8))
+    check(assert_type(p + das8, "PeriodSeries"), pd.Series)
```

Comment:
I think, but I'm not sure, that by putting this in quotes, you don't need to do the `if TYPE_CHECKING` import at the top.

Reply:
You do need it because it isn't a real class in `pandas` so pytest will error on import.